sdk-core/src/test/java/care/data4life/sdk/test/util/TestRxSchedulerRule.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk.test.util
+
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.TestScheduler
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class TestRxSchedulerRule : TestRule {
```

Comment:
Just a question, since I no info about the usage context - how this helps/how it supposed to be used?

Reply:
It could be added as a Rule to the test and will replace the Schedulers used by the RxJava calls by a TestScheduler.
You will be able to control the execution of RxJava either by `triggerActions()` or `advanceTimeBy(...)` manually. Helps to test RxJava.

Reply:
We have another TestSchedulerRule that is already in use but it's based on a Scheduler.trampoline() and changing that will affect a lot of tests.